src/builder.rs

```diff
@@ -52,4 +55,27 @@ impl<O: IsA<Builder>> BuilderExtManual for O {
             }
         }
     }
+
+    fn connect_signals_full<P: FnMut(&Builder, &str) -> Box<dyn Fn(&[glib::Value]) -> Option<glib::Value> + Send + Sync + 'static>>(&self, func: P) {
```

Comment:
I already see people complaining about the trait bounds on the closure :) `Send` and `Sync` are not needed here as this is all for GTK objects. Remove those bounds, and if necessary use some `unsafe` code to cast the bounds in there again for `object.connect()` :)